test/stats_test.go

```diff
@@ -32,91 +31,116 @@ import (
 	"google.golang.org/grpc/stats"
 )
 
-// TestPeerForClientStatsHandler tests the scenario where stats handler
-// (having peer as part of its struct) has peer enriched as part of
-// stream context.
+// TestPeerForClientStatsHandler configures a stats handler that
+// verifies that peer is sent for OutPayload, InPayload, End
+// stats handlers.
```

Comment:
Maybe all stats handler callouts instead of begin and picker updated?

Reply:
done